### ISSUE TITLE
parallelization: foreach -> BiocParallel

### ISSUE THREAD

**Issue by nhejazi** [nhejazi/methyvim#9]
In switching from a standard parallelized loop using `foreach` to the use of the same backend through `BiocParallel`, an odd (apparently low-level) error arises. In particular, it would be ideal to implement the parallelized loop using a combination of `future`, `doFuture`, and `BiocParallel` (see lines 205-218 of the `methyvim` function as of ef5332f).

When this loop is invoked, the resultant error and `traceback()` appear as follows:
```r
R >     methy_vim_out <- BiocParallel::bplapply(X = methy_tmle_ind,
...                                             FUN = methyvim_tmle,
...                                             methytmle_screened = methy_tmle,
...                                             var_of_interest = var_of_interest,
...                                             type = type,
...                                             corr = corr_max,
...                                             obs_per_covar = obs_per_covar,
...                                             target_param = "ate",
...                                             family = tmle_args$family,
...                                             g_lib = tmle_args$g_lib,
...                                             Q_lib = tmle_args$Q_lib,
...                                             return_ic = return_ic
...                                            )
Error in updt[[as.integer(i)]] <- .error(msg) :
  attempt to select more than one element in integerOneIndex
In addition: Warning message:
In updt[[as.integer(i)]] <- .error(msg) : NAs introduced by coercion
R >
R > traceback()
8: value[[3L]](cond)
7: tryCatchOne(expr, names, parentenv, handlers[[1L]])
6: tryCatchList(expr, classes, parentenv, handlers)
5: tryCatch({
       foreach::foreach(X = X, .errorhandling = handle) %dopar%
           FUN(X, ...)
   }, error = function(e) {
       txt <- "'DoparParam()' does not support partial results"
       updt <- rep(list(.error_not_available(txt)), length(X))
       msg <- conditionMessage(e)
       i <- sub("task ([[:digit:]]+).*", "\\1", msg)
       updt[[as.integer(i)]] <- .error(msg)
       updt
   })
4: bplapply(X, FUN, ..., BPREDO = BPREDO, BPPARAM = BPPARAM)
3: bplapply(X, FUN, ..., BPREDO = BPREDO, BPPARAM = BPPARAM)
2: BiocParallel::bplapply(X = methy_tmle_ind, FUN = methyvim_tmle,
       methytmle_screened = methy_tmle, var_of_interest = var_of_interest,
       type = type, corr = corr_max, obs_per_covar = obs_per_covar,
       target_param = "ate", family = tmle_args$family, g_lib = tmle_args$g_lib,
       Q_lib = tmle_args$Q_lib, return_ic = return_ic)
1: BiocParallel::bplapply(X = methy_tmle_ind, FUN = methyvim_tmle,
       methytmle_screened = methy_tmle, var_of_interest = var_of_interest,
       type = type, corr = corr_max, obs_per_covar = obs_per_covar,
       target_param = "ate", family = tmle_args$family, g_lib = tmle_args$g_lib,
       Q_lib = tmle_args$Q_lib, return_ic = return_ic)
```

**Comment by nhejazi** [comment]
This is not really a problem. `BiocParallel::bplapply` is complaining because several of the arguments are not defined outside of the calling function. The error arises when calling the chunk above in an interactive R session, but it should work just fine when invoked by `methyvim`.

**Comment by nhejazi** [comment]
Completely fixed, `foreach` totally removed as of 02a567f.
